Core/include/Acts/Propagator/EigenStepper.ipp

```diff
@@ -250,24 +250,20 @@ Acts::Result<double> Acts::EigenStepper<E, A>::step(
     // Furthermore, we're constructing K in place of J, and since
     // K₁₁ = I₈ = J₁₁ and K₂₁ = 0₈ = D₂₁, we don't actually need to touch those
     // sub-matrices at all!
-    if ((D.topLeftCorner<4, 4>().isIdentity()) &&
-        (D.bottomLeftCorner<4, 4>().isZero()) &&
-        (state.stepping.jacTransport.template topLeftCorner<4, 4>()
-             .isIdentity()) &&
-        (state.stepping.jacTransport.template bottomLeftCorner<4, 4>()
-             .isZero())) {
-      state.stepping.jacTransport.template topRightCorner<4, 4>() +=
-          D.topRightCorner<4, 4>() *
-          state.stepping.jacTransport.template bottomRightCorner<4, 4>();
-      state.stepping.jacTransport.template bottomRightCorner<4, 4>() =
-          (D.bottomRightCorner<4, 4>() *
-           state.stepping.jacTransport.template bottomRightCorner<4, 4>())
-              .eval();
-    } else {
-      // For safety purposes, we provide a full matrix multiplication as a
-      // backup strategy.
-      state.stepping.jacTransport = D * state.stepping.jacTransport;
-    }
+    assert((D.topLeftCorner<4, 4>().isIdentity()) &&
+           (D.bottomLeftCorner<4, 4>().isZero()) &&
+           (state.stepping.jacTransport.template topLeftCorner<4, 4>()
+                .isIdentity()) &&
+           (state.stepping.jacTransport.template bottomLeftCorner<4, 4>()
+                .isZero()));
```

Comment:
nitpick: should we make multiple asserts for this?

Reply:
Can do